acceptance/tests/agent/agent_with_waitforlock.rb

```diff
@@ -0,0 +1,26 @@
+test_name "agent with waitforlock"
+
+tag 'audit:high'
+
+step "agent without --waitforlock flag fails when a client is already in progress" do
+  agents.each do |agent|
+    # the first puppet agent -t command starts on one thread
+    # the seccond command after '&' sleeps 0.2 before staarting so the
+    # first command has time to start
+    on(agent, puppet('agent', "-t & sleep 0.2; puppet agent -t") ,:acceptable_exit_codes => [1]) do
+      assert_match(/Run of Puppet configuration client already in progress; skipping/,stdout)
+    end
```

Comment:
This seems racy and likely to cause transient failures, especially on platforms where the agent takes more than a non-trivial amount of time to load ruby files, run facter, acquire the agent run lock, etc. I think you might have better luck/consistency creating an integration to verify the locking behavior.